sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSink.scala

```diff
@@ -147,9 +147,13 @@ class FileStreamSink(
     if (batchId <= fileLog.getLatestBatchId().getOrElse(-1L)) {
       logInfo(s"Skipping already committed batch $batchId")
     } else {
+      // To avoid file name collision, we should generate a new job ID for every write job, instead
+      // of using batchId, as we may use the same batchId to write files again, if the streaming job
+      // fails and we restore from the checkpoint.
+      val jobId = java.util.UUID.randomUUID().toString
```

Comment:
One change of SPARK-33402 was including some timestamp/version info. That's potentially quite handy later just to see when things were created/order

Reply:
This is the job id for spark file commit protocol. In Hadoop `JobId`, we do append the timestamp info: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala#L266

But that's a different story.